pkg/stores/proxy/proxy_store_test.go

```diff
@@ -0,0 +1,104 @@
+package proxy
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/apiserver/pkg/types"
+	"github.com/rancher/steve/pkg/client"
+	"github.com/rancher/wrangler/pkg/schemas"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/sync/errgroup"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	schema2 "k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/fake"
+	"k8s.io/client-go/rest"
+	clientgotesting "k8s.io/client-go/testing"
+)
+
+var c *watch.FakeWatcher
+
+type testFactory struct {
+	*client.Factory
+
+	fakeClient *fake.FakeDynamicClient
+}
+
+func TestWatchNamesErrReceive(t *testing.T) {
+	testClientFactory, err := client.NewFactory(&rest.Config{}, false)
+	assert.Nil(t, err)
+
+	fakeClient := fake.NewSimpleDynamicClient(runtime.NewScheme())
+	c = watch.NewFakeWithChanSize(5, true)
+	defer c.Stop()
+	errMsgsToSend := []string{"err1", "err2", "err3"}
+	c.Add(&v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "testsecret1"}})
```

Comment:
Nit: 

Both `testsecret1` and `testsecret2` are part of the names we are watching. Would also be good to feed an event through with a name that we aren't watching for, to make sure `WatchName` doesn't return that object.

In addition, it would be good to return an object which errors when you call `meta.Accessor()` on it, to cover that case.

Marking as a nit due to priority, timeline, and that these are backfill cases.